### DISCUSSION TITLE
Having eui64 listed in child table

### DISCUSSION THREAD

**Discussion by tlifschitz** [openthread/openthread#9341] [Ideas]
Hi all!

In our application we uniquely identify each device by it factory assigned eui64. I can imagine we are not only the ones with this use case (eg [#9074 ](https://github.com/openthread/openthread/discussions/9074 ))

In order to perform some connectivity diagnostics and checks it would be useful to have the eui64 directrly available in the child table thus in the [meshdiag topology](https://github.com/openthread/openthread/tree/main/src/cli#meshdiag-topology-ip6-addrschildren) command (since it queries the child table of all routers in the network).

Of course an alternative would be maintain a _rloc_ to _eui64_ or _extaddress_ to _eui64_ mapping in the application layer and query the _eui64_ of every new unknown device. But the concern is of course minimizing traffic with the SEDs to avoid unnecessary power consumption. 

So adding the eui64 into the child table might be a good idea. I was thinking in this:

- Defining a new Mle Tlv to store the eui64 value
- Appending the eui64tlv to the message sent in [Mle::SendChildIdRequest](https://github.com/openthread/openthread/blob/499f6680a3afd600fdd8f2d003ea533334b4741f/src/core/thread/mle.cpp#L1777)
- Parse it in [MleRouter::HandleChildIdRequest](https://github.com/openthread/openthread/blob/499f6680a3afd600fdd8f2d003ea533334b4741f/src/core/thread/mle_router.cpp#L2061)
- The child is created in [MleRouter::SendChildIdResponse](https://github.com/openthread/openthread/blob/499f6680a3afd600fdd8f2d003ea533334b4741f/src/core/thread/mle_router.cpp#L2931C7-L2931C38)
- Now the child table has the eui64 of each child
- Handle this new value in all cli commands accordingly 

I invite everyone more familiar with the codebase to share their opinion whether this is a good idea or maybe there is a simpler / cleaner way.

Thank you!
Tobías

**Comment by jwhui** [comment]
Thread generally avoids using the IEEE EUI-64 since it is a static identifier.

What you proposed could work, but it involves modifying existing MLE messages and would not interoperate with other Thread devices that strictly conform to the Thread Specification.

An alternative approach would be to advertise server/service in the network data - it could simply encode an IPv6 address for the server/service. Devices joining the network would then parse the network data and register their presence with the service. This would be compatible with existing Thread devices.
